src/Course/ExactlyOne.hs

```diff
@@ -282,15 +295,23 @@ instance Eq a => Eq (ExactlyOne a) where
 --   c) Literally, what do you see when you ask `:type (==)` at the `ghci` prompt?
 --   d) Explain why there's a different between the signature you see when you ask `:info (==)` and the signature you see when you ask `:type (==)`.
 question_ExactlyOne_4 :: (P.String, P.String, P.String, P.String)
-question_ExactlyOne_4 = error "todo"
+question_ExactlyOne_4 =
+  ( "I see a typeclass defining the concept of equality, along with instances for all types that support this constraint."
+  , "`(==)` is a method defined in the typeclass `Eq a` taking two values of type `a` and returning a `Bool`"
+  , "`(==)` is a function from `a` to `a` to `Bool` in the typeclass of `Eq a`."
+  , "The signature of `Eq` in info is nested under the context of `class Eq a` so it is implicitly constrained, whereas the type info must explicitly include it since it does not nest the type inside of the type class."
+  )
 
 -- | Question ExactlyOne 4.5
 --
 -- Remove the instance context so that line 265 reads `instance Eq (ExactlyOne a) where`.
 --  a) What does the compiler tell you when you try to compile it?
 --  b) How does the compiler come to its conclusion? Is it being reasonable?
 question_ExactlyOne_4_5 :: (P.String, P.String)
-question_ExactlyOne_4_5 = error "todo"
+question_ExactlyOne_4_5 =
+  ( "There is a compile error coming from the use of `==` without knowing whether or not the type `a` supports that method (`a == a`)"
+  , "The compiler comes to this conclusion when resolving the expression `x == y` and searching for a valid method for the expression with `x :: a`, `y :: a`. In searching, it finds none and makes the valid suggestion to constrain the type `a` into `Eq a`. This is reasonable, since failing to do so would allow for undefined behavior for types that don't support this method."
```

Comment:
idk if you'll get undefined behavior. it's more like GHC can't compile because it just doesn't know what code to substitute in place of the `==` there.